timely/src/dataflow/channels/mod.rs

```diff
@@ -28,7 +28,12 @@ pub struct Message<T, D> {
 impl<T, D> Message<T, D> {
     /// Default buffer size.
     pub fn default_length() -> usize {
-        1024
+        const MESSAGE_BUFFER_SIZE: usize = 1 << 13;
+        match std::mem::size_of::<D>() {
+            0 => MESSAGE_BUFFER_SIZE,
```

Comment:
If it's a zst you may as well just use `usize::MAX`, vectors of zsts don't allocate anyways

Reply:
You're right! Changed the PR!

Reply:
As it turns out, `usize::MAX` leads to overflows in Timely's `to_stream` operator.